doc/data/messages/u/use-dict-literal/good.py

```diff
@@ -1 +1,7 @@
 empty_dict = {}
+
+# create using a literal dict
+new_dict = {"foo": "bar"}
+
+# cast the result of a method
+new_dict = dict(some_method())
```

Comment:
Is this better than ``dict(**another_dict)`` or ``another_dict.copy()`` ? We should probably justify this in details imo.

Reply:
Well I'd say `{…}` is always faster than `dict(…)`. So `dict(**another_dict)` should be refactored to `{**another_dict}` and that's what this message would suggest.

Which of those is better when copying a dict is probably out of scope here. `copy`, `deepcopy` and `{**another_dict}` are all accepted to be valid.

My goal with this example `new_dict = dict(some_method())` was to show that casting a value to a dict are considered ok in the `dict` constructor. Finding out if one of those could be refactored requires quite a bit more work.

In short: all `dict` calls that contain args will be ignored. `**kwargs` and `keyword=value` arguments are fine since we can make a suggestion to rewrite them as literals.

I hope this makes everything even more confusing! :stuck_out_tongue_winking_eye:

Reply:
Shouldn't we suggest this then ?
```suggestion
new_dict = {**another_dict}
```

Reply:
> Are you saying I should just list all examples that are valid for this specific pylint message?

No but the examples in good should reflect the example in bad (i.e. what is in good is "the official fix"). 

For ``dict(some_method()) `` we don't have a matching bad example because we don't know, so we could remove it.

Reply:
+1 for Pierre's last comment, I was also a bit confused how the third example in bad/good are connected (or not).
We should definitely add a 'good' example for copying a dict.

Reply:
Ah, now I get it, thanks!

I replace the `dict(some_method())` example with:

```py
# shallow copy a dict
new_dict = {**another_dict}
```